### PR TITLE
Disable some things

### PR DESCRIPTION
**Pull request by jkoan**
- Disable support for things that are not needed inside the android app
- Disable "mg" in general because its unmaintained

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -709,6 +709,13 @@ if(ANDROID)
    set_with_reason(vehicle/android "Android detected, NDK:${ANDROID_NDK_API_VERSION}, API:${ANDROID_API_VERSION}" TRUE)
    set(SHARED_LIBNAVIT TRUE)
 
+   # We have a mobile Device they dont need null components and most devices have limeted space, also download size is important 
+   set_with_reason(vehicle/null "Android detected, saveing space" FALSE)
```
Comment (pgrandin) [inline CMakeLists.txt]: `saveing` -> `saving`

File: CMakeLists.txt [hunk 2]
```diff
@@ -489,7 +489,7 @@ add_module(gui/internal "Default" TRUE)
 add_module(map/binfile "Default" TRUE)
 add_module(map/filter "Default" TRUE)
 if(NOT MSVC)
```
Comment (pgrandin) [inline CMakeLists.txt]: Why do we disable it also for Windows?

File: CMakeLists.txt [hunk 3]
```diff
@@ -709,6 +709,13 @@ if(ANDROID)
    set_with_reason(vehicle/android "Android detected, NDK:${ANDROID_NDK_API_VERSION}, API:${ANDROID_API_VERSION}" TRUE)
    set(SHARED_LIBNAVIT TRUE)
 
+   # We have a mobile Device they dont need null components and most devices have limeted space, also download size is important
```
Comment (pgrandin) [inline CMakeLists.txt]: `limeted` -> `limited`

**Comment by sleske** [comment]
Disabling map/mg and the "null" drivers for Android makes sense - I don't think anyone uses them. I would keep shapefile for now - it's only 164 kiB (or about 1% or the Navit APK). 

At any rate, I strongly object to globally disabling map/mg - either we keep building it, or we delete it. Keeping it in the source tree but not building it will probably cause it to break, so we might as well delete it.
Actually, deleting it might be the best option - while it is historically important (it was Navit's first map driver), the maps it can read are proprietary and no longer available (not to mention over ten years old). Or is there a reason to keep them?

**Comment by jkoan** [comment]
Hi @sleske thanks for your opinion. I will delete map/mg after this PR has been merged. Therefor i will open another PR. About Android im Okay. Update will come in a short time.

File: CMakeLists.txt (hunk 2 above)
Comment (jkoan) [reply CMakeLists.txt]: See the main discussion. All others are fixed

**Pull request closed without merging**